portfolio/src/main/java/com/google/sps/servlets/CommentService.java

```diff
@@ -0,0 +1,153 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ImagesServiceFailureException; 
+import com.google.appengine.api.images.ServingUrlOptions;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+import java.time.ZonedDateTime;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.servlet.http.HttpServletRequest;
+
+/** 
+ * This class contains the fields and methods
+ * required to use the Comment feature.
+ */
+ public final class CommentService {
+
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+ 
+  public List<Comment> getComments(PreparedQuery results, int limit) {
+    
+    // /data page is updated to contain JSON file of all images up to limit. 
+    List<Comment> comments = results.asList(FetchOptions.Builder.withLimit(limit))
+        .stream()
+        .map(entity -> {
+            
+            String postedTime = (String) entity.getProperty("postedTime");
+            String email = (String) entity.getProperty("email");
+            String title = (String) entity.getProperty("title");
+            String text = (String) entity.getProperty("text");
+            String imageUrl = (String) entity.getProperty("imageUrl");
+
+            return new Comment(postedTime, email, title, text, imageUrl);
+        })
+        .collect(Collectors.toList());
+    return comments;
+
+  }
+  
+  /** Adds the comment to the datastore. */
+  public void saveComment(HttpServletRequest request) {
```

Comment:
This method should accept a Comment instead of creating a dependency on Http here. It will make this method much easier to test.

Reply:
Done.